### PR TITLE
customcluster-controller-UT

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**

**What this PR does / why we need it**:
Do some Unit testing for customcluster_controller.go to vreify its function
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
Please change the title to be sentence

File: pkg/gclient/gclient.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+// aggregatedScheme aggregates Kubernetes and extended schemes.
+var aggregatedScheme = runtime.NewScheme()
+
+func init() {
+	utilruntime.Must(scheme.AddToScheme(aggregatedScheme))            // add Kubernetes schemes
+	utilruntime.Must(mcsv1alpha1.AddToScheme(aggregatedScheme))       // add mcs-api schemes
+	utilruntime.Must(clusterapiv1beta1.AddToScheme(aggregatedScheme)) // add cluster-api v1beta1 schemes
+	utilruntime.Must(v1alpha1.AddToScheme(aggregatedScheme))          // add v1alpha1-customMachine schemes
+	utilruntime.Must(controlplanev1.AddToScheme(aggregatedScheme))    // add controlplanev1 schemes
```
Comment (hzxuzhonghu) [inline pkg/gclient/gclient.go:39]: I donot think we should add it again, we have already registered scheme in cmd/cluster-operator/scheme/scheme.go

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+			Namespace: "default",
+			Name:      "test-pod",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Kind: CustomClusterKind,
+					Name: "test-customcluster",
+					UID:  "customcluster",
+				},
+			},
+		},
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:57]: nit: we can abstract a hepler function to construct a pod, which can be reused by all the tests below

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+	}
+	r := &CustomClusterController{
+		Client: fake.NewClientBuilder().WithScheme(gclient.NewSchema()).
+			WithRuntimeObjects(testCustomCluster, testCustomMachine, testKcp, testPod).
+			Build(),
+	}
+
+	result, err := r.reconcileDelete(ctx, testCustomCluster, testCustomMachine, testKcp)
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:264]: You can construct more cases to coverage more branches, go or goland can test the code coverage

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+				WithRuntimeObjects(testCustomCluster, testCustomMachine, testCluster, testKcp1).
+				Build(),
+		}
+
+		_, actual := r.reconcileProvision(ctx, testCustomCluster, testCustomMachine, testCluster, testKcp1)
+		_, expect := r.ensureClusterConfigCreated(ctx, testCustomCluster, testCustomCluster, testCluster, testKcp1)
+		assert.Equal(t, expect, actual)
+	})
+
+	t.Run("test reconcileProvision function", func(t *testing.T) {
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:356]: prefer table diven test mode

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+				Client: fake.NewClientBuilder().WithScheme(gclient.NewSchema()).
+					WithObjects(tc.customCluster, testCustomMachine, testCluster, testKcp).Build(),
+			}
+			_, actual := r.reconcile(ctx, tc.customCluster, testCustomMachine, testCluster, testKcp)
+			_, expect := r.reconcileProvision(ctx, tc.customCluster, testCustomMachine, testCluster, testKcp)
+			assert.Equal(t, expect, actual)
+		})
+	}
+
+	t.Run("reconcile test", func(t *testing.T) {
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:473]: move into `testCustomClusters`

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+
+	"kurator.dev/kurator/pkg/apis/infra/v1alpha1"
+	"kurator.dev/kurator/pkg/gclient"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
```
Comment (Xieql) [inline pkg/cluster-operator/customcluster_controller_test.go:34]: The package import order in your code is incorrect. It's generally recommended to follow the order of importing packages as: 
official libraries, (`"context"  "testing"`)
third-party libraries, , (the others)
and local libraries. (	`"kurator.dev/kurator/pkg/apis/infra/v1alpha1" "kurator.dev/kurator/pkg/gclient"`)
current import order can not pass lint check.

File: pkg/gclient/gclient.go (hunk 1 above)
Comment (LiZhenCheng9527) [reply pkg/gclient/gclient.go:39]: You are right. We dont need to add this file.

**Pull request closed without merging**